hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/profiling/DefaultConsumerProfiler.java

```diff
@@ -0,0 +1,77 @@
+package pl.allegro.tech.hermes.consumers.consumer.profiling;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.util.StopWatch;
+import pl.allegro.tech.hermes.api.SubscriptionName;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+public class DefaultConsumerProfiler implements ConsumerProfiler {
+
+    private static final Logger logger = LoggerFactory.getLogger(DefaultConsumerProfiler.class);
+
+    private final SubscriptionName subscriptionName;
+    private StopWatch stopWatch;
+    private final Map<String, StopWatch> partialMeasurements;
+    private long retryDelayMillis = 0;
```

Comment:
This is somewhat fragile that we rely on the fact that `saveRetryDelay` and `flushMeasurements` need to be called from the same thread in order to guarantee that `flushMeasurements` will be executed correctly. I'd say we need to at least document in comments if we don't want to make it volatile

Reply:
right, i have made it volatile